src/components/select/select.js

```diff
@@ -893,10 +893,33 @@ function SelectMenuDirective($parse, $mdUtil, $mdConstant, $mdTheming) {
       var newVal = self.isMultiple ? values : values[0];
       var prevVal = self.ngModel.$modelValue;
 
-      if (usingTrackBy ? !angular.equals(prevVal, newVal) : (prevVal + '') !== newVal) {
+      if (usingTrackBy ? !equals(prevVal, newVal) : (prevVal + '') !== newVal) {
         self.ngModel.$setViewValue(newVal);
         self.ngModel.$render();
       }
+
+      function equals(prevVal, newVal) {
+        if (self.isMultiple) {
+          if (!angular.isArray(prevVal)) {
+            // newVal is always an array when self.isMultiple is true
+            // thus, if prevVal is not an array they are different
+            return false;
+          } else if (prevVal.length !== newVal.length) {
+            // they are different if they have different length
+            return false;
+          } else {
+            // if they have the same length, then they are different
+            // if an item in the newVal array can't be found in the prevVal
+            return newVal.every(function(newValItem) {
+              return !!prevVal.find(function(prevValItem) {
+                return self.hashGetter(prevValItem) === self.hashGetter(newValItem);
```

Comment:
This could be made a little more efficient.
Now, you compute hashes `O(prevVal.length * newVal.length)`, while you only need to compute each hash once (`O(prevVal.length + newVal.length)`).